x-pack/plugin/security/qa/operator-privileges-tests/src/javaRestTest/java/org/elasticsearch/xpack/security/operator/OperatorPrivilegesIT.java

```diff
@@ -94,6 +95,39 @@ public void testOperatorUserCanCallNonOperatorOnlyApi() throws IOException {
         client().performRequest(mainRequest);
     }
 
+    public void testServiceAccountOperatorUserCanCallNonOperatorOnlyApi() throws IOException {
+        final Request mainRequest = new Request("GET", "/");
+        mainRequest.setOptions(
+            RequestOptions.DEFAULT.toBuilder()
+                .addHeader("Authorization", "Bearer AAEAAWVsYXN0aWMva2liYW5hL2tpYmFuYS10b2tlbjpndGw5dll2VlRMS2xjcWpHcEJRTWNn")
+        ); // elastic/kibana/kibana-token
+        client().performRequest(mainRequest);
+    }
+
+    public void testServiceAccountUpdateOperatorSettings() throws IOException {
+        final Map<String, Object> settings = new HashMap<>(
+            Map.of("xpack.security.http.filter.enabled", "false", "xpack.security.transport.filter.enabled", "false")
+        );
+
+        ResponseException responseException = expectThrows(ResponseException.class, () -> updateSettings(settings, null));
+        assertThat(responseException.getResponse().getStatusLine().getStatusCode(), equalTo(403));
+        assertThat(responseException.getMessage(), containsString("is unauthorized for user"));
+        assertThat(responseException.getMessage(), containsString("because it requires operator privileges"));
+        assertTrue(getPersistentSettings().isEmpty());
+
+        // call it with a service account that is listed as an operator. Ideally we would assert a success, but without additional plugins
+        // that change the operator rules, there are not any overlapping actions between service accounts and operator privileges
+        // so the best we can do is to assert that operator privileges are *not* the reason for the failure
+        responseException = expectThrows(
```

Comment:
I can also add a similar test to the internal project that does change the operator rules so we have more coverage that just the error message.